### DISCUSSION TITLE
Camera Intrinsics

### DISCUSSION THREAD

**Discussion by zichenwang01** [mitsuba-renderer/mitsuba3#1397] [Q&A] [answered]
Hi,

I am having a bit of trouble declaring a camera with the intrinsic matrix. 

If I understand correctly, the `to_world` transformation corresponds to the extrinsic matrix, but I will need to compute the `fov` and other parameters as an equivalent of the intrinsic matrix.

For example, if I have a camera with an image resolution 1600*1200 and an intrinsic matrix
```
[ [ 2.8923e+03, -5.5663e-04,  8.2321e+02,  0.0000e+00],
  [ 0.0000e+00,  2.8832e+03,  6.1907e+02,  0.0000e+00],
  [ 0.0000e+00,  0.0000e+00,  1.0000e+00,  0.0000e+00],
  [ 0.0000e+00,  0.0000e+00,  0.0000e+00,  1.0000e+00] ]
```

How should I declare such a camera in Mitsuba? Huge thanks in advance!

Edit: I get that I can compute 
```
fovx = 2 * math.atan(image_width / (2 * fx)) * (180 / math.pi)
fovy = 2 * math.atan(image_height / (2 * fy)) * (180 / math.pi),
```
but how should I specify `fovx, fovy` (as opposed to a single `fov`) and the optical center `cx, cy` in Mitsuba?

**Comment by jijiangli-swu-worker** [comment, marked as answer]
As I have known about mitsuba3, it use perspective projection matrix convert from 3d space to image plane, which means that unlike intrinsics matrix, it assums that `fx=fy` so if you specify `fov_x`, then you dont need to specify `fov_y`. `cx` and `cy` you can find in perspective camera plugin to specify `offset_principle_x` and `offset_principle_y` as a percentage rather than pixel as intrinsics

**Comment by merlinND** [comment]
Hello @zichenwang01,

Please have a look at the documentation of the `perspective` sensor, which describes the extrinsic (`to_world`) and intrinsic (`fov`, `fov_axis` / `focal_length`, `principal_point_offset_*`) parameters.
https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_sensors.html#perspective-pinhole-camera-perspective

**Comment by zichenwang01** [comment]
I see. Thanks!
